nexus/src/app/instance_network.rs

```diff
@@ -52,230 +47,6 @@ impl super::Nexus {
         Ok(boundary_switches)
     }
 
-    /// Ensures that V2P mappings exist that indicate that the instance with ID
-    /// `instance_id` is resident on the sled with ID `sled_id`.
-    pub(crate) async fn create_instance_v2p_mappings(
-        &self,
-        opctx: &OpContext,
-        instance_id: Uuid,
-        sled_id: Uuid,
-    ) -> Result<(), Error> {
-        info!(&self.log, "creating V2P mappings for instance";
-              "instance_id" => %instance_id,
-              "sled_id" => %sled_id);
-
-        // For every sled that isn't the sled this instance was allocated to, create
-        // a virtual to physical mapping for each of this instance's NICs.
-        //
-        // For the mappings to be correct, a few invariants must hold:
-        //
-        // - mappings must be set whenever an instance's sled changes (eg.
-        //   during instance creation, migration, stop + start)
-        //
-        // - an instances' sled must not change while its corresponding mappings
-        //   are being created
-        //
-        // - the same mapping creation must be broadcast to all sleds
-        //
-        // A more targeted approach would be to see what other instances share
-        // the VPC this instance is in (or more generally, what instances should
-        // have connectivity to this one), see what sleds those are allocated
-        // to, and only create V2P mappings for those sleds.
-        //
-        // There's additional work with this approach:
-        //
-        // - it means that delete calls are required as well as set calls,
-        //   meaning that now the ordering of those matters (this may also
-        //   necessitate a generation number for V2P mappings)
```

Comment:
I think it mostly depends on whether or not it is acceptable to send / read the entire configuration during a reconciliation loop. For things like NAT, the total configuration can be quite large, so we employ generation numbers so we can send only the required updates instead of the full set. For v2p mappings, the individual messages are much smaller. My napkin math from before showed that if we had a vm per core, with a vnic per vm, that would be something in the ballpark of 213KB for a full set of updates for a single sled, which seems reasonable.

Reply:
:+1: